### PR TITLE
fix(browser): wait for compilation before watch readiness

### PR DESCRIPTION
**Pull request by 9aoy**
## Motivation

After #1816, browser watch mode can report an empty initial run before compilation and watcher registration finish. Adding a test immediately after readiness can then fail with `Cannot find module`, as seen in Ubuntu e2e runs on main.

## Changes

Wait for browser watch compilation to settle before reporting readiness, including additional compilation passes, while keeping browser launch concurrent. Propagate compilation failures and clean up the browser and servers when startup fails.

Add programmatic e2e coverage for delayed empty startup, an additional compilation pass, adding the first test, and startup failure cleanup. Local browser unit tests and targeted default, CommonJS, and threads e2e tests passed, along with type and lint checks. The [full CI matrix](https://github.com/web-infra-dev/rstest/actions/runs/34435421249) has been manually triggered; results are pending.

### CODE REVIEW

File: packages/browser/src/browserRsbuild.ts [hunk 1]
```diff
@@ -1987,10 +2040,17 @@ export const createBrowserRuntime = async ({
       headless: forceHeadless ?? browserLaunchOptions.headless,
       providerOptions: browserLaunchOptions.providerOptions,
     });
+    browser = runtime.browser;
+    let generation;
+    do {
+      generation = buildGeneration;
+      await Promise.all(waitForBuilds.map((waitForBuild) => waitForBuild()));
```
[the remark this answers is not in the document]
Comment (9aoy) [reply packages/browser/src/browserRsbuild.ts:2047]: This looks like a valid edge case, but we expect it to be uncommon in typical usage: a test file must be added during startup, after entry discovery and before the watch session is ready. We'll leave this as a known limitation for now and revisit it if users report encountering it.
